sim/battle.ts

```diff
@@ -3442,6 +3441,9 @@ export class Battle extends Dex.ModdedDex {
 	}
 }
 
-function newUnimplementedError(name: string) {
-	return new Error(`The ${name} function needs to be implemented in scripts.js or the battle format.`);
+class UnimplementedError extends Error {
+	constructor(name: string) {
```

Comment:
I vaguely remember there being some trick you have to do to make Error subclasses work correctly.

Reply:
> We can inherit from Error and other built-in error classes normally, just need to take care of name property and don’t forget to call super.

https://javascript.info/custom-errors

[MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error#Custom_Error_Types) talks about `Error.captureStackTrace` on Node, but then also links to [Stack Overflow](https://stackoverflow.com/questions/1382107/whats-a-good-way-to-extend-error-in-javascript), which seems to agree that in ES6 the way the first site I quoted above explained things is correct. (but given how its not immediately obvious what the 'correct' way is, its perhaps not surprising I originally punted by just making this a `function`...).

Anyway, went ahead and threw in the `Error.captureStackTrace` check.

Reply:
See my comment above: ``Error.captureStackTrace()`` is no longer needed.